replication/static/src/main/java/io/deephaven/replicators/TupleCodeGenerator.java

```diff
@@ -67,14 +67,14 @@ public class TupleCodeGenerator {
-         DOUBLE( "Double",  double.class, LHS + " == " + RHS                              ,    "Double.hashCode(" + VAL + ")", "DoubleComparisons.compare(" + LHS + ", " + RHS + ")", OUT + ".writeDouble(" + VAL + ')', IN + ".readDouble()", "io.deephaven.util.compare.DoubleComparisons"                     ),
-           CHAR(   "Char",    char.class, LHS + " == " + RHS                              , "Character.hashCode(" + VAL + ")", "CharComparisons.compare("   + LHS + ", " + RHS + ")", OUT + ".writeChar("   + VAL + ')', IN + ".readChar()"  , "io.deephaven.util.compare.CharComparisons"                       ),
-         OBJECT( "Object",  Object.class, "ObjectComparisons.eq(" + LHS + ", " + RHS + ")",   "Objects.hashCode(" + VAL + ")", "ObjectComparisons.compare(" + LHS + ", " + RHS + ")", OUT + ".writeObject(" + VAL + ')', IN + ".readObject()", "io.deephaven.util.compare.ObjectComparisons", "java.util.Objects"),
+           BYTE(   "Byte",    byte.class, LHS + " == " + RHS                                ,      "Byte.hashCode(" + VAL + ")", "ByteComparisons.compare("   + LHS + ", " + RHS + ")", OUT + ".writeByte("   + VAL + ')', IN + ".readByte()"  , "io.deephaven.util.compare.ByteComparisons"                       ),
+          SHORT(  "Short",   short.class, LHS + " == " + RHS                                ,     "Short.hashCode(" + VAL + ")", "ShortComparisons.compare("  + LHS + ", " + RHS + ")", OUT + ".writeShort("  + VAL + ')', IN + ".readShort()" , "io.deephaven.util.compare.ShortComparisons"                      ),
+            INT(    "Int",     int.class, LHS + " == " + RHS                                ,   "Integer.hashCode(" + VAL + ")", "IntComparisons.compare("    + LHS + ", " + RHS + ")", OUT + ".writeInt("    + VAL + ')', IN + ".readInt()"   , "io.deephaven.util.compare.IntComparisons"                        ),
+           LONG(   "Long",    long.class, LHS + " == " + RHS                                ,      "Long.hashCode(" + VAL + ")", "LongComparisons.compare("   + LHS + ", " + RHS + ")", OUT + ".writeLong("   + VAL + ')', IN + ".readLong()"  , "io.deephaven.util.compare.LongComparisons"                       ),
+          FLOAT(  "Float",   float.class, "FloatComparisons.eq(" + LHS + ", " + RHS + ")"   ,     "Float.hashCode(" + VAL + ")", "FloatComparisons.compare("  + LHS + ", " + RHS + ")", OUT + ".writeFloat("  + VAL + ')', IN + ".readFloat()" , "io.deephaven.util.compare.FloatComparisons"                      ),
+         DOUBLE( "Double",  double.class, "DoubleComparisons.eq(" + LHS + ", " + RHS + ")"  ,    "Double.hashCode(" + VAL + ")", "DoubleComparisons.compare(" + LHS + ", " + RHS + ")", OUT + ".writeDouble(" + VAL + ')', IN + ".readDouble()", "io.deephaven.util.compare.DoubleComparisons"                     ),
```

Comment:
I think this also brings up the point that the hashcode is incorrect too; it should be using FloatComparisons.hashCode / DoubleComparisons.hashCode. You can demonstrate the inconsistent eq / hashcode behavior w/ a tuple that contains 0.0 and another that contains -0.0.